### PR TITLE
docs: Add Thanos sidecar Prometheus HTTP client config format

### PR DESCRIPTION
**Pull request by heylongdacoder**
Fixes: #5462

Signed-off-by: heylongdacoder <heylongdacoder@gmail.com>

* [ ] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

### CODE REVIEW

**Review by fpetkovski — APPROVED** [review]
Nice 👍

File: docs/components/sidecar.md [hunk 1]
```diff
@@ -181,3 +183,36 @@ Flags:
       --version                  Show application version.
 
 ```
+
+## Configuration
+
+### Prometheus HTTP client
+
+The `--prometheus.http-client` and `--prometheus.http-client-file` flags allows the configuration for Prometheus HTTP client.
```
Comment (douglascamata) [inline docs/components/sidecar.md:191]: It's worth mentioning that the 1st flag expects the YAML file contents and the 2nd expects a path to the YAML file.

**Review by douglascamata — CHANGES_REQUESTED** [review]
I think we need a small addition to the docs to make it clearer, otherwise pretty good job. 👍

File: docs/components/sidecar.md (hunk 1 above)
Comment (heylongdacoder) [reply docs/components/sidecar.md]: done!

**Review by douglascamata — APPROVED** [review]
LGTM :shipit: 👍

**Review by yeya24 — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**